pingora-core/src/services/listening.rs

```diff
@@ -163,14 +163,19 @@ impl<A: ServerApp + Send + Sync + 'static> Service<A> {
             };
             match new_io {
                 Ok(io) => {
+                    let peer_addr = io.peer_addr();
```

Comment:
nit, personally would prefer moving this into the `async move` closure below just before the handshake, but that's very minor.